pages/fundamentals/data-types.mdx

```diff
@@ -772,6 +773,180 @@ and display it nicely in the client-side results.
+
+The point.distance function returns the distance between two points. Distance can only be calculated between points of the same type.
+For WGS-84 points the distance is calculated in meters and is calculated using the [Haversine](https://en.wikipedia.org/wiki/Haversine_formula) formula.
+For WGS-84 3D points the Haversine formula is applied to longitude and latitude and then the Pythagorean theorem is applied between the calculated distance and the height.
+For Cartesian points, the Pythagorean theorem is applied.
+If any of the arguments of the point.distance function evaluates to null or the point types don't match then the function returns null.
+
+{<h5> Usage: </h5>}
+
+Example of the distance between London and Zagreb:
+```cypher
+  WITH point({longitude: 51.51, latitude: -0.12}) as london,
+       point({longitude: 45.82, latitude: 15.97}) as zagreb
+  RETURN point.distance(london, zagreb) AS result;
+```
+```plaintext
++----------------------------+
+| result                     |
++----------------------------+
+| 1.88621e+06                |
++----------------------------+
+```
+
+Example of the distance between two points in 2D Cartesian space:
+```cypher
+  RETURN point.distance(
+              point({x: 0, y: 0}),
+              point({x: 3, y: 3})
+            ) AS result;
+```
+```plaintext
++----------------------------+
+| result                     |
++----------------------------+
+| 4.24264                    |
++----------------------------+
+```
+
+
+#### point.withinbbox(point: [null, point], lower_left_corner: [null, point], upper_right_corner: [null, point]) -> [null, boolean]
+
+The point.withinbbox function returns whether the given `point` is inside or on the edge of the bounding box created by `lower_left_corner` and `upper_right_corner`.
+For WGS-84 points this also works if crossing the the international date line.
+
+{<h5> Usage: </h5>}
+
+Example of a point inside the bounding box:
+```cypher
+  RETURN point.withinbbox(
+              point({x:   0, y:  0}),
+              point({x:  -1, y: -1}),
+              point({x:   1, y:  1})
+            ) AS result;
+```
+```plaintext
++----------------------------+
+| result                     |
++----------------------------+
+| true                       |
++----------------------------+
+```
+
+Example of crossing the international date line:
+```cypher
+  RETURN point.withinbbox(
+              point({longitude:  180, latitude: 58.0}),
+              point({longitude:  179, latitude: 57.0}),
+              point({longitude: -179, latitude: 59.0})
+            ) AS result;
+```
+```plaintext
++----------------------------+
+| result                     |
++----------------------------+
+| true                       |
++----------------------------+
+```
+
+If any of the arguments of the point.withinbbox function evaluates to null or the point types don't match then the function returns null.
+
 ## Procedures API
```

Comment:
Is the API extended with this?

Reply:
What do you mean? These are builtin functions, not query modules

Reply:
I was referring to point type, if smth needs to be done with that

Reply:
Its supported in mgconsole and neo4j driver since we followed the bolt protocol. It doesn't exist in mgp yet.